docs/advanced/tls_hardware_acceleration.md

```diff
@@ -0,0 +1,131 @@
+
+1. If you want to replace functions in the ECP module, you need to implement the mandatory utility functions:
+    - these are functions that do not have a counterpart in the standard mbed TLS implementation and their only purpose is to facilitate the integration of the accelerated functions.
+    - `mbedtls_internal_ecp_grp_capable`: implement it to tell mbed TLS if the cryptographic hardware can handle the group. If the answer is no, then mbed TLS will fall back to the software implementation to continue the operation.
+    - `mbedtls_internal_ecp_init` and `mbedtls_internal_ecp_free` are optional. Use them to optimise if you are replacing a function in the ECP module.
+    - For more information about the utility functions read the subsection about the [ECP](#How-to-implement-ECP-module-functions) module.
+
+1. Implement the selected functions with the help of your hardware accelerator and copy the source code to the `features/mbedtls/targets/TARGET_XXXX` directory specific to your target. You may create a directory structure similar to the one you have for the HAL if you feel it appropriate. These functions have the same name as the ones they replace. We have [doxygen documentation for the original functions](https://tls.mbed.org/api/). The exception to the naming conventions is the ECP module, where an internal API is exposed to enable hardware acceleration. These functions too have a doxygen documentation and you can find them in the `ecp_internal.h` header file.
+
+1. Since mbed TLS is a static link library you also have to somehow notify the compiler/linker that the alternative implementations are present. To do this, you have to set the macros corresponding to the selected functions. You can read more on this in the [subsection about setting macros](#How-to-set-the-macros?) section.
+
+### How to implement ECP module functions
+
+mbed TLS supports only curves over prime fields and uses mostly curves of short Weierstrass form. The function `ecp_add_mixed_alt` and the functions having `_jac_` in their names are related to point arithmetic on curves in short Weierstrass form. The only Montgomery curve supported is Curve25519. To accelerate operations on this curve you have to replace the three functions with `_mxz_` in their name.
+
+Hardware accelerators may support different kinds of elliptic curves. You may need to tell some of these accelerators what kind of curve operation it has to perform by setting some registers or calling some special instructions. If performing this takes significant amount of time or power, then you may not want mbed TLS to do this step unnecessarily. Unfortunately the replaceable functions in this module are relatively low level, and it may not be necessary to do this initialisation and de-initialisation in each of them.
```

Comment:
The language you're using thinks of the acceleration as a 'thing', but actually it may come in many forms, such as another crypto library, or SIMD instructions. Can I suggest the following phrasing:

'The method of accelerating the ECP module may support different kinds of elliptic curves. If that acceleration is a hardware accelerator, you may need to indicate what kind of curve operation the accelerator has to perform by setting a registers or executing a special instruction.'